graphqlapi/src/main/java/io/stargate/graphql/schema/cqlfirst/dml/fetchers/BulkMutationFetcher.java

```diff
@@ -65,7 +66,15 @@ protected BulkMutationFetcher(
       buildException = e;
     }
     OperationDefinition operation = environment.getOperationDefinition();
-    if (operation.getDirectives().stream().anyMatch(d -> d.getName().equals(ATOMIC_DIRECTIVE))
+
+    // if the caller expects the atomic mutations,
+    // he should not use fire-and-forget behavior provided by the async directive
```

Comment:
Is this not what logged batches provide?

Reply:
yes, this is why we allow logged batch (`atomic` directive) or the new `async`.

Reply:
Yes, but why not allow both? If the client does not want to wait for the acknowledgement of a batch write, it's the client's choice. Why would we want to explicitly forbid it?

Reply:
+1 an async batch makes sense to me.

Reply:
Here is my reasoning around forbidding it:

On the one hand, we may allow both operations - it seems that batch without waiting for the `ack`. On the other hand, I think that we should enforce good patterns and prohibit the usage of harmful patterns. 

I assume that when the client annotated its operations with `@atomic`, he wants a form of atomic "transactions". In that case, I think that the caller SHOULD wait for the ack. `Atomic` without `ack` does not give any value to the caller. It is tough when there are N operations in one `Atomic` update. How will the caller validate if this batch was indeed executed correctly? He would need to perform N GET queries for all mutations in the `atomic` batch and wait for the result. Sure, it is feasible but far from ideal.
When prohibiting `async` and `atomic` we are enforcing good usage patterns of our clients. 

Please note that `atomic` =! `bulk` inserts. The current implementation allows bulk inserts (N records) with `@async` directive. Wdyt?The accepted field is relevant and fulfilled with data only when used with the @%s directive.
wdyt?

Reply:
I think there is bunch of use cases where I don't care about the result of the atomic operation.. So `@atomic` & `@async` makes sense for me from the developer point of view.. It's his choice at the end..

Reply:
It seems that I am out-voted here :D 
Implemented `async` AND `atomic`:
https://github.com/stargate/stargate/pull/940/commits/49746aca90892df8674f9409bb8632e8bc077d52

Reply:
> I believe C* provides atomic batches in the sense that all of none of the mutations in the batch take effect eventually